src/test/java/com/connexta/search/index/IndexServiceImplTest.java

```diff
@@ -0,0 +1,254 @@
+    when(mockRequestHeadersSpec.retrieve()).thenReturn(mockResponseSpec);
+    when(mockResponseSpec.onStatus(any(Predicate.class), any(Function.class)))
+        .thenReturn(mockResponseSpec2);
+    when(mockResponseSpec2.bodyToMono(Resource.class)).thenReturn(mockMono);
+    when(mockMono.block()).thenReturn(mockResource);
+
+    final IOException ioException = new IOException();
+    when(mockResource.getInputStream()).thenThrow(ioException);
+
+    final SearchException thrown =
+        assertThrows(
+            SearchException.class,
+            () ->
+                ((IndexService) new IndexServiceImpl(mockSearchManager, mockWebClient))
+                    .index(datasetId, irmUri));
+    assertThat(thrown.getStatus(), is(HttpStatus.BAD_REQUEST));
+    assertThat(thrown.getCause(), is(ioException));
+  }
+
+  @ParameterizedTest
+  @MethodSource("exceptionsThrownSearchManager")
+  void testSearchManagerExceptions(final RuntimeException runtimeException) throws Exception {
+    final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
+
+    final RuntimeException thrown;
+
+    try (final MockWebServer storeMockWebServer = new MockWebServer()) {
+      storeMockWebServer.start();
+      storeMockWebServer.enqueue(
+          new MockResponse()
+              .setBody(
+                  String.format(
+                      "{ \"%s\" : \"All the color had been leached from Winterfell until only grey and white remained.\" }",
+                      EXT_EXTRACTED_TEXT))
+              .setResponseCode(200));
+      final URI irmUri = storeMockWebServer.url(String.format("/dataset/%s/irm", datasetId)).uri();
+      doThrow(runtimeException)
+          .when(mockSearchManager)
+          .index(eq(datasetId), eq(irmUri), any(InputStream.class));
+
+      thrown =
+          assertThrows(
+              RuntimeException.class,
+              () ->
+                  ((IndexService) new IndexServiceImpl(mockSearchManager, WebClient.create()))
+                      .index(datasetId, irmUri));
+    }
+
+    assertThat(
+        "thrown exception is the exact same exception thrown by the SearchManager",
+        thrown,
+        is(runtimeException));
+  }
+
+  @Test
+  void testIndex() throws Exception {
+    final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
+    final String body =
+        String.format(
+            "{ \"%s\" : \"All the color had been leached from Winterfell until only grey and white remained.\" }",
+            EXT_EXTRACTED_TEXT);
+
+    final URI irmUri;
+    try (final MockWebServer storeMockWebServer = new MockWebServer()) {
+      storeMockWebServer.start();
+      storeMockWebServer.enqueue(new MockResponse().setBody(body).setResponseCode(200));
+      irmUri = storeMockWebServer.url(String.format("/dataset/%s/irm", datasetId)).uri();
+
+      ((IndexService) new IndexServiceImpl(mockSearchManager, WebClient.create()))
+          .index(datasetId, irmUri);
+    }
+
+    //    verify(mockSearchManager)
```

Comment:
❓ Remove comment?

Reply:
I actually need help with this. I want to verify that the input stream contains the irm contents. Do you know why `StringUtils.equals(IOUtils.toString(argument, StandardCharsets.UTF_8), body)` doesn't work?